src/DvidManager.jsx

```diff
@@ -7,52 +7,105 @@ import PropTypes from 'prop-types';
   initialized = 0;
 
   onInitCompleted = undefined;
 
-  segmentationURL = 'https://flyem.dvid.io/d925633ed0974da78e2bb5cf38d01f4d/segmentation';
+  grayscaleURL = 'dvid://https://flyem.dvid.io/ab6e610d4fe140aba0e030645a1d7229/grayscalejpeg';
+
+  segmentationURL = 'dvid://https://flyem.dvid.io/d925633ed0974da78e2bb5cf38d01f4d/segmentation';
 
   init = (onInitCompleted) => {
     if (this.initialized === 0) {
+      if (this.localStorageAvailable()) {
+        // If localStorage is available, use it to remember the URLs across sessions.
+        const g = localStorage.getItem(KEY_GRAYSCALE_SOURCE);
+        if (g) {
+          this.grayscaleURL = g;
+        }
+        const s = localStorage.getItem(KEY_SEGMENTATION_SOURCE);
+        if (s) {
+          this.segmentationURL = s;
+        }
+      }
       this.onInitCompleted = onInitCompleted;
       this.initialized = 1;
     }
   }
 
+  grayscaleSourceURL = () => (
+    this.grayscaleURL
+  )
+
+  segmentationSourceURL = () => (
+    this.segmentationURL
+  )
+
+  sparseVolSize = (bodyId, onCompletion) => {
+    const url = `${this.segmentationAPIURL()}/sparsevol-size/${bodyId}`;
+    fetch(url).then((response) => response.json()).then((data) => onCompletion(data));
+  }
+
+  bodyId = (bodyPt, onCompletion) => {
+    const key = `${bodyPt[0]}_${bodyPt[1]}_${bodyPt[2]}`;
+    const url = `${this.segmentationAPIURL()}/label/${key}`;
+    fetch(url).then((response) => response.json()).then((data) => onCompletion(data.Label));
+  };
+
+  //
+
   onDialogClosed = () => {
     this.onInitCompleted();
   }
 
-  grayscaleSourceURL = () => {
-    // TODO: Get this value from the proper UI.
-    const result = 'dvid://https://flyem.dvid.io/ab6e610d4fe140aba0e030645a1d7229/grayscalejpeg';
-    return (result);
+  onGrayscaleSourceChange = (event) => {
+    this.grayscaleURL = event.target.value;
+    if (this.localStorageAvailable()) {
+      localStorage.setItem(KEY_GRAYSCALE_SOURCE, this.grayscaleURL);
+    }
   }
 
-  segmentationSourceURL = () => {
-    // TODO: Get this value from the proper UI.
-    const result = `dvid://${this.segmentationURL}`;
-    return (result);
+  onSegmentationSourceChange = (event) => {
+    this.segmentationURL = event.target.value;
+    if (this.localStorageAvailable()) {
+      localStorage.setItem(KEY_SEGMENTATION_SOURCE, this.segmentationURL);
+    }
   }
 
   segmentationAPIURL = () => {
-    const parts = this.segmentationURL.split('/');
+    const i = this.segmentationURL.indexOf('http');
+    const url = this.segmentationURL.substring(i);
+    const parts = url.split('/');
     return (`${parts[0]}//${parts[2]}/api/node/${parts[3]}/${parts[4]}`);
   }
 
-  sparseVolSize = (bodyId, onCompletion) => {
-    const url = `${this.segmentationAPIURL()}/sparsevol-size/${bodyId}`;
-    fetch(url).then((response) => response.json()).then((data) => onCompletion(data));
+  localStorageAvailable = () => {
```

Comment:
@hubbardp, Seems like this would better as a utility function that gets imported, rather than a class function. I added /src/utils/ on my auth branch as somewhere to put these types of functions. It looks generally useful and I would probably use it in other locations.

Reply:
I guess.  But my feeling is that `fetch` already factors out so much functionality that the generalized part of the code here is just a one-liner.

Reply:
This comment was supposed to reference the localStorageAvailable function. Bad placement on my part